pkg/proxy/ipvs/graceful_termination_test.go

```diff
@@ -403,18 +404,28 @@ func Test_GracefulDeleteRS(t *testing.T) {
 }
 
 func Test_RaceTerminateRSList(t *testing.T) {
-	ipvs := &utilipvstest.FakeIPVS{}
+	ipvs := utilipvstest.NewFake()
 	gracefulTerminationManager := NewGracefulTerminationManager(ipvs)
 
+	// run in parallel to cause the race
 	go func() {
 		for i := 1; i <= 10; i++ {
-			for j := 1; i <= 100; j++ {
-				gracefulTerminationManager.rsList.add(makeListItem(i, j))
+			for j := 1; j <= 100; j++ {
+				item := makeListItem(i, j)
+				gracefulTerminationManager.rsList.add(item)
 			}
 		}
 	}()
 
-	if !gracefulTerminationManager.rsList.flushList(gracefulTerminationManager.deleteRsFunc) {
+	// wait until the list has some elements
+	for gracefulTerminationManager.rsList.len() < 20 {
+	}
+
+	// fake the handler to avoid the check against the IPVS virtual servers
+	fakeHandler := func(rsToDelete *listItem) (bool, error) {
```

Comment:
the other handler was doing some checks against the ipvs objects and failing, so it never tried to delete anything form the list, this test doesn't need that checks

Reply:
in the test, we only test race and don't care other thing

Reply:
yeah, but is racy and sometimes it gets to delete the list and other no